python/docs/source/migration_guide/pyspark_3.3_to_3.4.rst

```diff
@@ -0,0 +1,23 @@
+..  Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+..    http://www.apache.org/licenses/LICENSE-2.0
+
+..  Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+=================================
+Upgrading from PySpark 3.3 to 3.4
+=================================
+
+* In Spark 3.4, the schema of an array column is inferred by merging the schemas of all elements in the array. To restore the previous behavior where the schema is only inferred from the first element, set the spark configuration as follows: ``{"spark.sql.pyspark.legacy.inferArrayTypeFromFirstElement.enabled": True}``.
```

Comment:
I think we can just say set `spark.sql.pyspark.legacy.inferArrayTypeFromFirstElement.enabled` to `true` (just to be consistent with other places like https://github.com/apache/spark/blob/a6b207e00ac8add3d6b98a7f4a9f5e3406c4704a/python/docs/source/migration_guide/pyspark_3.1_to_3.2.rst#upgrading-from-pyspark-31-to-32

Reply:
[7cde786](https://github.com/apache/spark/commit/7cde786d3cf40e6cfbfbd42cff513d94b2003c06)